src/common/libschedutil/ops.c

```diff
@@ -197,49 +212,55 @@ static void service_unregister (flux_t *h)
     flux_future_destroy (f);
 }
 
-struct ops_context *schedutil_ops_register (flux_t *h,
-                                            op_alloc_f *alloc_cb,
-                                            op_free_f *free_cb,
-                                            op_exception_f *exception_cb,
-                                            void *arg)
+int schedutil_ops_register (schedutil_t *s,
+                            op_alloc_f *alloc_cb,
+                            op_free_f *free_cb,
+                            op_exception_f *exception_cb,
+                            void *arg)
 {
     struct ops_context *ctx;
 
-    if (!h || !alloc_cb || !free_cb || !exception_cb) {
+    if (!s || !alloc_cb || !free_cb || !exception_cb) {
         errno = EINVAL;
-        return NULL;
+        return -1;
     }
     if (!(ctx = calloc (1, sizeof (*ctx))))
```

Comment:
Should the ops function pointers just be part of the `schedutil_t` struct and the separate `struct ops_context` eliminated?  If users always register ops, then maybe they should just be required arguments to `schedutil_create()`?

Reply:
:+1: I had the same comment above, sounds like this would be the way to go...

Reply:
>  If users always register ops, then maybe they should just be required arguments to schedutil_create()?

Yeah. I wasn't sure if users would always register ops or not.  I think that makes sense to add them as arguments to `schedutil_create`.

Reply:
Added in https://github.com/flux-framework/flux-core/pull/2226/commits/73cefce93334e32378af106bda2017d30d1caa23